clients/client-shell/cmds/d2g/d2g.go

```diff
@@ -179,3 +207,8 @@ func validateJSON(input []byte, schema string) error {
 
 	return nil
 }
+
+func isYAMLExtension(fileExtension string) bool {
```

Comment:
As mentioned above, this is not the best way to determine the format of the content. Furthmore, this function would consider a file called `foo.notyaml` to be a yaml file. The standard library provides [filepath.Ext](https://pkg.go.dev/path/filepath#Ext) for this purpose which handles variable length extensions, but we shouldn't need to consider the filename, especially because the input can come from standard in, so we need a means to be able to deduce it without a filename anyway.